---
layout:     post
title:      RNA-seq分析流程整理
subtitle:   水稻PB1+Pi9的RNA-seq分析流程（仅使用部分数据作示范）
date:       2018-10-15
author:     DL
header-img: img/home-bg-o.jpg
catalog: true
tags:
    - 转录组入门和进阶
---

> 正所谓前人栽树，后人乘凉。
> 
> 本文参考来源：《RNA-seq数据分析实用方法》、《生物信息学计算技术和软件导论》、《新一代测序数据分析》及本人自己的RNA-seq分析过程。
> 


# 一、数据准备
	双端测序的fastq文件：control_1.fastq、control_2.fastq、treatment_1.fastq、treatment_2.fastq（举例）。

	全基因组文件：genome_xxx.fa

	gtf注释文件：genome_xxx.gtf

---

# 二、分析流程
## 方法一：
2.1 **建立数据分析目录:**

在分析之前，在当前工作目录下建立三个文件夹及相应的文件，文件结构如下：

	annotation文件夹存放水稻基因组的注释文件；
	
	genome文件夹存放水稻的基因组序列文件及其相应索引文件；

	data文件夹存放水稻在两种不同条件下测序的RNA-Seq数据，每个条件下只有一个样品，没有重复。

2.2 **准备数据和软件:**

（1）软件准备：
	
	Tophat2（v2.0.12）：短读段映射。
	
	Bowtie2（v2.2.3）：一个超快的短读段比对工具。
	
	Cufflinks（v2.0.0）：转录组装配。
	
	HTseq：用于NGS数据分析的python脚本。
	
	DESeq：R的bioconductor包进行差异表达分析。
	
	Samtools（v0.1.18）：读段分析工具。
	
	CummeRbund：R的bioconductor包进行差异表达可视化分析。
	
（2）sra解压为fastq：

	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588125.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588126.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588127.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588128.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588129.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588130.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588131.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588132.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588133.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588134.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588135.sra
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/sratoolkit.2.8.2-centos_linux64/bin/fastq-dump --split-3 SRR3588136.sra

3.**查看数据质量并进行过滤:**

less SRR3588125_1.fq  (查看一下fastq格式数据内容)
（1）使用fastqc软件查看数据质量：

首先选择以下两组数据：

	~/PB1+Pi9/fastqc/fastqc  SRR3588125_1.fq

	~/PB1+Pi9/fastqc/fastqc  SRR3588125_2.fq

会生成 SRR3588125_1.fq_fastqc SRR3588125_2.fq_fastqc这两个文件夹

（2）数据过滤：用Trimmomatic

（3）再次用fastqc查看数据质量

4.**Bowtie2建立参考基因组的索引:**
	
	[wwkong@localhost genome-data]$ /home/wwkong/programs/bowtie2-2.2.4/bowtie2-build Oryza_sativa.37.dna.fa Oryza_sativa.37.dna

输入该命令后，文件夹内会产生：xxx.1.bt2、xxx.2.bt2、xxx.3.bt2、xxx.4.bt2、xxx.rev.1.bt2、xxx.rev.2.bt2

接下来，建立软链接：

	[wwkong@localhost PB1+Pi9]$ ln -s /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf
	[wwkong@localhost PB1+Pi9]$ ln -s /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.dna.1.bt2
	[wwkong@localhost PB1+Pi9]$ ln -s /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.dna.2.bt2
	[wwkong@localhost PB1+Pi9]$ ln -s /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.dna.3.bt2
	[wwkong@localhost PB1+Pi9]$ ln -s /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.dna.4.bt2
	[wwkong@localhost PB1+Pi9]$ ln -s /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.dna.fa
	[wwkong@localhost PB1+Pi9]$ ln -s /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.dna.rev.1.bt2
	[wwkong@localhost PB1+Pi9]$ ln -s /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.dna.rev.2.bt2


5.**reads mapping到参考基因组——基于Bowtie2的Tophat2:**

	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-1 Oryza_sativa.37.dna SRR3588125_1.fastq SRR3588125_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-2 Oryza_sativa.37.dna SRR3588126_1.fastq SRR3588126_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-3 Oryza_sativa.37.dna SRR3588127_1.fastq SRR3588127_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-4 Oryza_sativa.37.dna SRR3588128_1.fastq SRR3588128_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-5 Oryza_sativa.37.dna SRR3588129_1.fastq SRR3588129_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-6 Oryza_sativa.37.dna SRR3588130_1.fastq SRR3588130_2.fastq

	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-M.oryzae-1 Oryza_sativa.37.dna SRR3588131_1.fastq SRR3588131_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-M.oryzae-2 Oryza_sativa.37.dna SRR3588132_1.fastq SRR3588132_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-M.oryzae-3 Oryza_sativa.37.dna SRR3588133_1.fastq SRR3588133_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-M.oryzae-4 Oryza_sativa.37.dna SRR3588134_1.fastq SRR3588134_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-M.oryzae-5 Oryza_sativa.37.dna SRR3588135_1.fastq SRR3588135_2.fastq
	[wwkong@localhost PB1+Pi9]$ /home/wwkong/programs/tophat-2.1.1.Linux_x86_64/tophat -p 8 -G Oryza_sativa.37.gtf --library-type fr-unstranded --no-discordant --no-convert-bam --keep-fasta-order -o PB1+Pi9-M.oryzae-6 Oryza_sativa.37.dna SRR3588136_1.fastq SRR3588136_2.fastq

执行以上命令后，会生成目录，里面会产生accepted_hits.sam文件。

6.**转录本组装——Cufflinks：Cufflinks是一套拼接转录本、定量表达量的软件（分为有参注释和无参注释两种）:**

	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-1/PB1+Pi9-1_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-1/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-2/PB1+Pi9-2_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-2/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-3/PB1+Pi9-3_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-3/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-4/PB1+Pi9-4_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-4/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-5/PB1+Pi9-5_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-5/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-6/PB1+Pi9-6_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-6/accepted_hits.sam

	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-1/PB1+Pi9-M.oryzae-1_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-1/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-2/PB1+Pi9-M.oryzae-2_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-2/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-3/PB1+Pi9-M.oryzae-3_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-3/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-4/PB1+Pi9-M.oryzae-4_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-4/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-5/PB1+Pi9-M.oryzae-5_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-5/accepted_hits.sam
	[wwkong@localhost PB1+Pi9]$ cufflinks -p 8 -o /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-6/PB1+Pi9-M.oryzae-6_clout  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-6/accepted_hits.sam

执行以上命令后，会在各自目录下生成transcript.gtf。

7.**在工作目录创建一个assemblies.txt：**
	
	[wwkong@localhost PB1+Pi9]$ vim assemblies.txt

该文本文件内应该含有以下内容：

	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-1/PB1+Pi9-1_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-2/PB1+Pi9-2_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-3/PB1+Pi9-3_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-4/PB1+Pi9-4_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-5/PB1+Pi9-5_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-6/PB1+Pi9-6_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-1/PB1+Pi9-M.oryzae-1_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-2/PB1+Pi9-M.oryzae-2_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-3/PB1+Pi9-M.oryzae-3_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-4/PB1+Pi9-M.oryzae-4_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-5/PB1+Pi9-M.oryzae-5_clout/transcripts.gtf
	/home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-6/PB1+Pi9-M.oryzae-6_clout/transcripts.gtf

8.**合并装配后的转录本文件（Cuffmerge）:**

	[wwkong@localhost PB1+Pi9]$ cuffmerge -g Oryza_sativa.37.gtf -s Oryza_sativa.37.dna.fa -p 8 assemblies.txt

9.**差异表达分析（Cuffdiff）**

10.**使用cummeRbund包进行差异表达分析**

## 方法二：

**1-6步完全一样。**

7.**转录本差异表达分析——htseq-count 和 deseq：分析差异表达基因的工具**

（1）htseq-count：

	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-1/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PB1+Pi9-TRAN-1.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-2/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PB1+Pi9-TRAN-2.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-3/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PB1+Pi9-TRAN-3.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-4/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PB1+Pi9-TRAN-4.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-5/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PB1+Pi9-TRAN-5.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-6/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PB1+Pi9-TRAN-6.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-1/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PPB1+Pi9-M-TRAN-1.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-2/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PPB1+Pi9-M-TRAN-2.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-3/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PPB1+Pi9-M-TRAN-3.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-4/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PPB1+Pi9-M-TRAN-4.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-5/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PPB1+Pi9-M-TRAN-5.count
	[wwkong@localhost PB1+Pi9]$ htseq-count  -m  intersection-strict  -i  transcript_id  -s  no  /home/wwkong/rice-RNA-Seq/Mor-GSE81906/PB1+Pi9/PB1+Pi9-M.oryzae-6/accepted_hits.sam  /home/wwkong/rice-RNA-Seq/genome-data/Oryza_sativa.37/Oryza_sativa.37.gtf > PPB1+Pi9-M-TRAN-6.count

（2）deseq：

	library("DESeq")
	muta1=read.table("PB1+Pi9-M-TRAN-1.count",row.names=1)
	muta2=read.table("PB1+Pi9-M-TRAN-2.count",row.names=1)
	muta3=read.table("PB1+Pi9-M-TRAN-3.count",row.names=1)
	muta4=read.table("PB1+Pi9-M-TRAN-4.count",row.names=1)
	muta5=read.table("PB1+Pi9-M-TRAN-5.count",row.names=1)
	muta6=read.table("PB1+Pi9-M-TRAN-6.count",row.names=1)

	wild1=read.table("PB1+Pi9-TRAN-1.count",row.names=1)
	wild2=read.table("PB1+Pi9-TRAN-2.count",row.names=1)
	wild3=read.table("PB1+Pi9-TRAN-3.count",row.names=1)
	wild4=read.table("PB1+Pi9-TRAN-4.count",row.names=1)
	wild5=read.table("PB1+Pi9-TRAN-5.count",row.names=1)
	wild6=read.table("PB1+Pi9-TRAN-6.count",row.names=1)

	data=cbind(muta1,muta2,muta3,muta4,muta5,muta6,wild1,wild2,wild3,wild4,wild5,wild6) 
	head(data)
	colnames(data)=c("Pi9M1","Pi9M2","Pi9M3","Pi9M4","Pi9M5","Pi9M6","Pi91","Pi92","Pi93","Pi94","Pi95","Pi96")
	head(data)
	write.table(data,file="data",quote=F)
	counts <- read.table ("data",sep="",header=TRUE)
	cor_table = cor (counts)
	write.table(cor_table,"PB1+Pi9-PB1+Pi9M-cor.data")
	conditions=c(rep("muta", 6),rep("wild", 6))
	cds <- newCountDataSet(counts, conditions)
	cds <- estimateSizeFactors(cds)
	sizeFactors(cds)
	cds <- estimateDispersions(cds, method = "per-condition", sharingMode="maximum")
	dispTable(cds)
	et <- nbinomTest(cds, "muta", "wild")
	write.table(et,file = "PB1+Pi9-PB1+Pi9M.DESeq.out",sep = "\t")


**方法二的补充：**

1.如果是pair-end数据，则必须按照reads名称排序后，再使用htseq-count：
	
	samtools sort -n accepted_hits.sam hits_name.sorted
	htseq-count -f bam --strand=no hit_name.sorted.bam Oryza_sativa.37.gtf > PB1+Pi9-TRAN-1.count

2.htseq-count：-i参数默认为gene_id，transcript_id由于模型问题，计数效果不佳。


