---
layout:     post
title:      2021-06-30-专业英语的Abstract
subtitle:   
date:       2021-06-30
author:     DL
header-img: img/home-bg-o.jpg
catalog: true
tags:
    - 文献精读
---

### 1.The promise of synthetic biology

**Abstract** 

&emsp;&emsp;DNA synthesis has become one of the technological bases of a new concept in biology: synthetic biology. The vision of synthetic biology is a systematic, hierarchical design of artificial, biology-inspired systems using robust, standardized, and well-characterized building blocks. The design concept and examples from four fields of application (genetic circuits, protein design, platform technologies, and pathway engineering) are discussed, which demonstrate the usefulness and the promises of synthetic biology. The vision of synthetic biology is to develop complex systems by simplified solutions using available material and knowledge. Synthetic biology also opens a door toward new biomaterials that do not occur in nature.


> [全文链接](https://pubmed.ncbi.nlm.nih.gov/17269152/)

---

### 2.Two "what if" experiments

**Abstract** 



> [全文链接]()

---

### 3.Building an artificial regulatory system to understand a natural one

**Abstract** 



> [全文链接]()

---

### 4.Turning lambda Cro into a transcriptional activator

**Abstract** 

&emsp;&emsp;According to our present understanding, **lambda repressor** bound to DNA stimulates transcription by touching RNA polymerase bound at an adjacent promoter. The **part** of repressor required for **activation** was identified in part by the **isolation of mutants** specifically impaired in transcriptional activation. The amino acids of repressor altered in these "positive control" mutants lie in an **acidic patch** on the surface of repressor that is closely apposed to RNA polymerase. In this study, we show that this **"activating patch" of repressor is sufficient for transcriptional activation in another sequence context**. We transfer this activating patch onto the surface of lambda Cro, a protein normally unable to activate transcription, and show that the modified Cro is a transcriptional activator. In addition, we provide evidence that the repressor protein of phage 434 also activates transcription using an activating patch similar to that of lambda repressor.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/2968842/)

---

### 5.Improving the Lac System for Synthetic Biology

**Abstract** 

&emsp;&emsp;

> [全文链接]()

---

### 6.De novo designed proteins from a library of artificial sequences function in Escherichia coli and enable cell growth

**Abstract** 

&emsp;&emsp;A central challenge of synthetic biology is to enable the growth of living systems using parts that are **not derived from nature**, but **designed and synthesized in the laboratory**. As an initial step toward achieving this goal, we probed the ability of **a collection of >10(6) de novo designed proteins** to provide biological functions necessary to **sustain cell growth**. Our collection of proteins was drawn from a combinatorial library of 102-residue sequences, designed by binary patterning of polar and nonpolar residues to fold into stable 4-helix bundles. We probed the capacity of proteins from this library to function in vivo by testing their abilities to **rescue 27 different knockout strains** of Escherichia coli, each deleted for a conditionally essential gene. Four different strains--ΔserB, ΔgltA, ΔilvA, and Δfes--were rescued by specific sequences from our library. Further experiments demonstrated that a strain simultaneously deleted for all four genes was rescued by co-expression of four novel sequences. Thus, cells deleted for ∼0.1% of the E. coli genome (and ∼1% of the genes required for growth under nutrient-poor conditions) can be sustained by sequences designed de novo.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/21245923/)

---

### 7.Directed evolution of recombinase specificity by split gene reassembly

**Abstract** 

&emsp;&emsp;The **engineering of new enzymes** that efficiently and specifically **modify DNA sequences** is necessary for the development of enhanced gene therapies and genetic studies. To address this need, we developed a robust strategy for **evolving site-specific recombinases with novel substrate specificities**. In this system, recombinase variants are selected for activity on new substrates based on enzyme-mediated reassembly of the gene encoding beta-lactamase that confers ampicillin resistance to Escherichia coli. This stringent evolution method was used to alter the specificities of catalytic domains in the context of a modular zinc finger-recombinase fusion protein. Gene reassembly was detectable over several orders of magnitude, which allowed for tunable selectivity and exceptional sensitivity. Engineered recombinases were evolved to react with sequences from the human genome with only three rounds of selection. Many of the evolved residues, selected from a randomly-mutated library, were conserved among other members of this family of recombinases. This enhanced evolution system will translate recombinase engineering and genome editing into a practical and expedient endeavor for academic, industrial and clinical applications.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/20194120/)

---

### 8.Environmental signal integration by a modular AND gate

**Abstract** 

&emsp;&emsp;Microorganisms use **genetic circuits** to integrate environmental information. We have **constructed a synthetic AND gate** in the bacterium **Escherichia coli** that integrates information from two promoters as inputs and activates a promoter output only when both input promoters are transcriptionally active. The integration occurs via an interaction between an mRNA and tRNA. The **first promoter** controls the transcription of a T7 RNA polymerase gene with two internal amber stop codons blocking translation. The **second promoter** controls the amber suppressor tRNA supD. When both components are transcribed, T7 RNA polymerase is synthesized and this in turn activates a T7 promoter. Because inputs and outputs are promoters, the design is modular; that is, it can be reconnected to integrate different input signals and the output can be used to drive different cellular responses. We demonstrate this modularity by wiring the gate to integrate natural promoters (responding to Mg(2+) and AI-1) and using it to implement a phenotypic output (invasion of mammalian cells). A mathematical model of the transfer function is derived and parameterized using experimental data.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/17700541/)

---

### 9.A synthetic oscillatory network of transcriptional regulators

**Abstract** 

&emsp;&emsp;Networks of interacting biomolecules carry out many essential functions in living cells, but the 'design principles' underlying the **functioning of such intracellular networks** remain poorly understood, despite intensive efforts including quantitative analysis of relatively simple systems. Here we present a complementary approach to this problem: the design and construction of a synthetic network to implement a particular function. We used three transcriptional repressor systems that are not part of any natural biological clock to build an oscillating network, termed the repressilator, in Escherichia coli. The network periodically induces the synthesis of green fluorescent protein as a readout of its state in individual cells. The resulting oscillations, with typical periods of hours, are slower than the cell-division cycle, so the state of the oscillator has to be transmitted from generation to generation. This artificial clock displays noisy behaviour, possibly because of stochastic fluctuations of its components. Such 'rational network design may lead both to the engineering of new cellular behaviours and to an improved understanding of naturally occurring networks.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/10659856/)

---

### 10.Construction of a genetic toggle switch in Escherichia coli

**Abstract** 

&emsp;&emsp;It has been proposed' that gene-regulatory circuits with virtually any desired property can be constructed from networks of simple regulatory elements. These properties, which include multistability and oscillations, have been found in specialized gene circuits such as the bacteriophage lambda switch and the Cyanobacteria circadian oscillator. However, these behaviours have not been demonstrated in networks of non-specialized regulatory components. Here we present the construction of a genetic toggle switch-a synthetic, bistable gene-regulatory network-in Escherichia coli and provide a simple theory that predicts the conditions necessary for bistability. The toggle is constructed from any two repressible promoters arranged in a mutually inhibitory network. It is flipped between stable states using transient chemical or thermal induction and exhibits a nearly ideal switching threshold. As a practical device, the toggle switch forms a synthetic, addressable cellular memory unit and has implications for biotechnology, biocomputing and gene therapy.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/10659857/)

---

### 11.Reprogramming Cellular Behavior with RNA Controllers Responsive to Endogenous Proteins

**Abstract** 

&emsp;&emsp;**Synthetic genetic devices** that interface with native cellular **pathways** can be used to change natural networks to implement new forms of control and behavior. The engineering of gene networks has been limited by an inability to interface with native components. We describe a class of RNA control devices that overcome these limitations by coupling increased abundance of particular proteins to targeted gene expression events through the regulation of alternative RNA splicing. We engineered RNA devices that detect signaling through the nuclear factor κB and Wnt signaling pathways in human cells and rewire these pathways to produce new behaviors, thereby linking disease markers to noninvasive sensing and reprogrammed cellular fates. Our work provides a genetic platform that can build programmable sensing-actuation devices enabling autonomous control over cellular behavior.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/21109673/)

---

### 12.The Case for RNA

**Abstract** 

&emsp;&emsp;

> [全文链接]()

---

### 13.Nascent transcript sequencing visualizes transcription at nucleotide resolution

**Abstract** 

&emsp;&emsp;Recent studies of transcription have revealed a level of complexity not previously appreciated even a few years ago, both in the intricate use of post-initiation control and the mass production of rapidly degraded transcripts. Dissection of these pathways requires strategies for precisely following transcripts as they are being produced. Here we present an approach (native elongating transcript sequencing, NET-seq), based on deep sequencing of 3' ends of nascent transcripts associated with RNA polymerase, to monitor transcription at nucleotide resolution. Application of NET-seq in Saccharomyces cerevisiae reveals that although promoters are generally capable of divergent transcription, the Rpd3S deacetylation complex enforces strong directionality to most promoters by suppressing antisense transcript initiation. Our studies also reveal pervasive polymerase pausing and backtracking throughout the body of transcripts. Average pause density shows prominent peaks at each of the first four nucleosomes, with the peak location occurring in good agreement with in vitro biophysical measurements. Thus, nucleosome-induced pausing represents a major barrier to transcriptional elongation in vivo.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/21248844/)

---

### 14.Measuring the activity of BioBrick promoters using an in vivo reference standard

**Abstract** 

Background: The engineering of many-component, synthetic biological systems is being made easier by the development of collections of reusable, standard biological parts. However, the complexity of biology makes it difficult to predict the extent to which such efforts will succeed. As a first practical example, the Registry of Standard Biological Parts started at MIT now maintains and distributes thousands of BioBrick standard biological parts. However, BioBrick parts are only standardized in terms of how individual parts are physically assembled into multi-component systems, and most parts remain uncharacterized. Standardized tools, techniques, and units of measurement are needed to facilitate the characterization and reuse of parts by independent researchers across many laboratories.

Results: We found that the absolute activity of BioBrick promoters varies across experimental conditions and measurement instruments. We choose one promoter (BBa_J23101) to serve as an in vivo reference standard for promoter activity. We demonstrated that, by measuring the activity of promoters relative to BBa_J23101, we could reduce variation in reported promoter activity due to differences in test conditions and measurement instruments by approximately 50%. We defined a Relative Promoter Unit (RPU) in order to report promoter characterization data in compatible units and developed a measurement kit so that researchers might more easily adopt RPU as a standard unit for reporting promoter activity. We distributed a set of test promoters to multiple labs and found good agreement in the reported relative activities of promoters so measured. We also characterized the relative activities of a reference collection of BioBrick promoters in order to further support adoption of RPU-based measurement standards.

Conclusion: Relative activity measurements based on an in vivoreference standard enables improved measurement of promoter activity given variation in measurement conditions and instruments. These improvements are sufficient to begin to support the measurement of promoter activities across many laboratories. Additional in vivo reference standards for other types of biological functions would seem likely to have similar utility, and could thus improve research on the design, production, and reuse of standard biological parts.



> [全文链接](https://pubmed.ncbi.nlm.nih.gov/19298678/)

---

### 15.A modular positive feedback-based gene amplifier

**Abstract** 

Background: Positive feedback is a common mechanism used in the regulation of many gene circuits as it can amplify the response to inducers and also generate binary outputs and hysteresis. In the context of electrical circuit design, positive feedback is often considered in the design of amplifiers. Similar approaches, therefore, may be used for the design of amplifiers in synthetic gene circuits with applications, for example, in cell-based sensors.

Results: We developed a modular positive feedback circuit that can function as a genetic signal amplifier, heightening the sensitivity to inducer signals as well as increasing maximum expression levels without the need for an external cofactor. The design utilizes a constitutively active, autoinducer-independent variant of the quorum-sensing regulator LuxR. We experimentally tested the ability of the positive feedback module to separately amplify the output of a one-component tetracycline sensor and a two-component aspartate sensor. In each case, the positive feedback module amplified the response to the respective inducers, both with regards to the dynamic range and sensitivity.

Conclusions: The advantage of our design is that the actual feedback mechanism depends only on a single gene and does not require any other modulation. Furthermore, this circuit can amplify any transcriptional signal, not just one encoded within the circuit or tuned by an external inducer. As our design is modular, it can potentially be used as a component in the design of more complex synthetic gene circuits.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/20187959/)

---

### 16.A fast, robust and tunable synthetic gene oscillator

**Abstract** 

&emsp;&emsp;One defining goal of synthetic biology is the development of engineering-based approaches that enable the construction of gene-regulatory networks according to 'design specifications' generated from computational modelling. This approach provides a systematic framework for exploring how a given regulatory network generates a particular phenotypic behaviour. Several fundamental gene circuits have been developed using this approach, including toggle switches and oscillators, and these have been applied in new contexts such as triggered biofilm development and cellular population control. Here we describe an engineered genetic oscillator in Escherichia coli that is fast, robust and persistent, with tunable oscillatory periods as fast as 13 min. The oscillator was designed using a previously modelled network architecture comprising linked positive and negative feedback loops. Using a microfluidic platform tailored for single-cell microscopy, we precisely control environmental conditions and monitor oscillations in individual cells through multiple cycles. Experiments reveal remarkable robustness and persistence of oscillations in the designed circuit; almost every cell exhibited large-amplitude fluorescence oscillations throughout observation runs. The oscillatory period can be tuned by altering inducer levels, temperature and the media source. Computational modelling demonstrates that the key design principle for constructing a robust oscillator is a time delay in the negative feedback loop, which can mechanistically arise from the cascade of cellular processes involved in forming a functional transcription factor. The positive feedback loop increases the robustness of the oscillations and allows for greater tunability. Examination of our refined model suggested the existence of a simplified oscillator design without positive feedback, and we construct an oscillator strain confirming this computational prediction.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/18971928/)

---

### 17.Programming cells by multiplex genome engineering and accelerated evolution

**Abstract** 

&emsp;&emsp;The breadth of genomic diversity found among organisms in nature allows populations to adapt to diverse environments. However, genomic diversity is difficult to generate in the laboratory and new phenotypes do not easily arise on practical timescales. Although in vitro and directed evolution methods have created genetic variants with usefully altered phenotypes, these methods are limited to laborious and serial manipulation of single genes and are not used for parallel and continuous directed evolution of gene networks or genomes. Here, we describe multiplex automated genome engineering (MAGE) for large-scale programming and evolution of cells. MAGE simultaneously targets many locations on the chromosome for modification in a single cell or across a population of cells, thus producing combinatorial genomic diversity. Because the process is cyclical and scalable, we constructed prototype devices that automate the MAGE technology to facilitate rapid and continuous generation of a diverse set of genetic changes (mismatches, insertions, deletions). We applied MAGE to optimize the 1-deoxy-D-xylulose-5-phosphate (DXP) biosynthesis pathway in Escherichia coli to overproduce the industrially important isoprenoid lycopene. Twenty-four genetic components in the DXP pathway were modified simultaneously using a complex pool of synthetic DNA, creating over 4.3 billion combinatorial genomic variants per day. We isolated variants with more than fivefold increase in lycopene production within 3 days, a significant improvement over existing metabolic engineering techniques. Our multiplex approach embraces engineering in the context of evolution by expediting the design and evolution of organisms with new and improved properties.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/19633652/)

---

### 18.A programmable dual-RNA-guided DNA endonuclease in adaptive bacterial immunity.

**Abstract** 

&emsp;&emsp;Clustered regularly interspaced short palindromic repeats (CRISPR)/CRISPR-associated (Cas) systems provide bacteria and archaea with adaptive immunity against viruses and plasmids by using CRISPR RNAs (crRNAs) to guide the silencing of invading nucleic acids. We show here that in a subset of these systems, the mature crRNA that is base-paired to trans-activating crRNA (tracrRNA) forms a two-RNA structure that directs the CRISPR-associated protein Cas9 to introduce double-stranded (ds) breaks in target DNA. At sites complementary to the crRNA-guide sequence, the Cas9 HNH nuclease domain cleaves the complementary strand, whereas the Cas9 RuvC-like domain cleaves the noncomplementary strand. The dual-tracrRNA:crRNA, when engineered as a single RNA chimera, also directs sequence-specific Cas9 dsDNA cleavage. Our study reveals a family of endonucleases that use dual-RNAs for site-specific DNA cleavage and highlights the potential to exploit the system for RNA-programmable genome editing.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/22745249/)

---

### 19.RNA-Guided Human Genome Engineering via Cas9

**Abstract** 

&emsp;&emsp;Bacteria and archaea have evolved adaptive immune defenses, termed clustered regularly interspaced short palindromic repeats (CRISPR)/CRISPR-associated (Cas) systems, that use short RNA to direct degradation of foreign nucleic acids. Here, we engineer the type II bacterial CRISPR system to function with custom guide RNA (gRNA) in human cells. For the endogenous AAVS1 locus, we obtained targeting rates of 10 to 25% in 293T cells, 13 to 8% in K562 cells, and 2 to 4% in induced pluripotent stem cells. We show that this process relies on CRISPR components; is sequence-specific; and, upon simultaneous introduction of multiple gRNAs, can effect multiplex editing of target loci. We also compute a genome-wide resource of ~190 K unique gRNAs targeting ~40.5% of human exons. Our results establish an RNA-guided editing tool for facile, robust, and multiplexable human genome engineering.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/23287722/)

---

### 20.Multiplex genome engineering using CRISPR/Cas systems.

**Abstract** 

&emsp;&emsp;Functional elucidation of causal genetic variants and elements requires precise genome editing technologies. The type II prokaryotic CRISPR (clustered regularly interspaced short palindromic repeats)/Cas adaptive immune system has been shown to facilitate RNA-guided site-specific DNA cleavage. We engineered two different type II CRISPR/Cas systems and demonstrate that Cas9 nucleases can be directed by short RNAs to induce precise cleavage at endogenous genomic loci in human and mouse cells. Cas9 can also be converted into a nicking enzyme to facilitate homology-directed repair with minimal mutagenic activity. Lastly, multiple guide sequences can be encoded into a single CRISPR array to enable simultaneous editing of several sites within the mammalian genome, demonstrating easy programmability and wide applicability of the RNA-guided nuclease technology.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/23287718/)

---

### 21.Single-step assembly of a gene and entire plasmid from large numbers of oligodeoxyribonucleotides

**Abstract** 

&emsp;&emsp;Here, we describe assembly PCR as a method for the synthesis of long DNA sequences from large numbers of oligodeoxyribonucleotides (oligos). The method, which is derived from DNA shuffling [Stemmer, Nature 370 (1994a) 389-391], does not rely on DNA ligase but instead relies on DNA polymerase to build increasingly longer DNA fragments during the assembly process. A 1.1-kb fragment containing the TEM-1 beta-lactamase-encoding gene (bla) was assembled in a single reaction from a total of 56 oligos, each 40 nucleotides (nt) in length. The synthetic gene was PCR amplified and cloned in a vector containing the tetracycline-resistance gene (TcR) as the sole selectable marker. Without relying on ampicillin (Ap) selection, 76% of the TcR colonies were ApR, making this approach a general method for the rapid and cost-effective synthesis of any gene. We tested the range of assembly PCR by synthesizing, in a single reaction vessel containing 134 oligos, a high-molecular-mass multimeric form of a 2.7-kb plasmid containing the bla gene, the alpha-fragment of the lacZ gene and the pUC origin of replication. Digestion with a unique restriction enzyme, followed by ligation and transformation in Escherichia coli, yielded the correct plasmid. Assembly PCR is well suited for several in vitro mutagenesis strategies.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/7590320/)

---

### 22.Study of messenger RNA inactivation and protein degradation in an Escherichia coli cell-free expression system

**Abstract** 

Background: A large amount of recombinant proteins can be synthesized in a few hours with Escherichia coli cell-free expression systems based on bacteriophage transcription. These cytoplasmic extracts are used in many applications that require large-scale protein production such as proteomics and high throughput techniques. In recent years, cell-free systems have also been used to engineer complex informational processes. These works, however, have been limited by the current available cell-free systems, which are not well adapted to these types of studies. In particular, no method has been proposed to increase the mRNA inactivation rate and the protein degradation rate in cell-free reactions. The construction of in vitro informational processes with interesting dynamics requires a balance between mRNA and protein synthesis (the source), and mRNA inactivation and protein degradation (the sink).

Results: Two quantitative studies are presented to characterize and to increase the global mRNA inactivation rate, and to accelerate the degradation of the synthesized proteins in an E. coli cell-free expression system driven by the endogenous RNA polymerase and sigma factor 70. The E. coli mRNA interferase MazF was used to increase and to adjust the mRNA inactivation rate of the Firefly luciferase (Luc) and of the enhanced green fluorescent protein (eGFP). Peptide tags specific to the endogenous E. coli AAA + proteases were used to induce and to adjust the protein degradation rate of eGFP. Messenger RNA inactivation rate, protein degradation rate, maturation time of Luc and eGFP were measured.

Conclusions: The global mRNA turnover and the protein degradation rate can be accelerated and tuned in a biologically relevant range in a cell-free reaction with quantitative procedures easy to implement. These features broaden the capabilities of cell-free systems with a better control of gene expression. This cell-free extract could find some applications in new research areas such as in vitro synthetic biology and systems biology where engineering informational processes requires a quantitative control of mRNA inactivation and protein degradation.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/20594314/)

---

### 23.Emergent properties of reduced-genome Escherichia coli.

**Abstract** 

&emsp;&emsp;With the use of synthetic biology, we reduced the Escherichia coli K-12 genome by making planned, precise deletions. The multiple-deletion series (MDS) strains, with genome reductions up to 15%, were designed by identifying nonessential genes and sequences for elimination, including recombinogenic or mobile DNA and cryptic virulence genes, while preserving good growth profiles and protein production. Genome reduction also led to unanticipated beneficial properties: high electroporation efficiency and accurate propagation of recombinant genes and plasmids that were unstable in other strains. Eradication of stress-induced transposition evidently stabilized the MDS genomes and provided some of the new properties.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/16645050/)

---

### 24.Rewiring the specificity of two-component signal transduction systems

**Abstract** 

&emsp;&emsp;Two-component signal transduction systems are the predominant means by which bacteria sense and respond to environmental stimuli. Bacteria often employ tens or hundreds of these paralogous signaling systems, comprised of histidine kinases (HKs) and their cognate response regulators (RRs). Faithful transmission of information through these signaling pathways and avoidance of detrimental crosstalk demand exquisite specificity of HK-RR interactions. To identify the determinants of two-component signaling specificity, we examined patterns of amino acid coevolution in large, multiple sequence alignments of cognate kinase-regulator pairs. Guided by these results, we demonstrate that a subset of the coevolving residues is sufficient, when mutated, to completely switch the substrate specificity of the kinase EnvZ. Our results shed light on the basis of molecular discrimination in two-component signaling pathways, provide a general approach for the rational rewiring of these pathways, and suggest that analyses of coevolution may facilitate the reprogramming of other signaling systems and protein-protein interactions.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/18555780/)

---

### 25.Use of modular, synthetic scaffolds for improved production of glucaric acid in engineered E. coli

**Abstract** 

&emsp;&emsp;The field of metabolic engineering has the potential to produce a wide variety of chemicals in both an inexpensive and ecologically-friendly manner. Heterologous expression of novel combinations of enzymes promises to provide new or improved synthetic routes towards a substantially increased diversity of small molecules. Recently, we constructed a synthetic pathway to produce d-glucaric acid, a molecule that has been deemed a "top-value added chemical" from biomass, starting from glucose. Limiting flux through the pathway is the second recombinant step, catalyzed by myo-inositol oxygenase (MIOX), whose activity is strongly influenced by the concentration of the myo-inositol substrate. To synthetically increase the effective concentration of myo-inositol, polypeptide scaffolds were built from protein-protein interaction domains to co-localize all three pathway enzymes in a designable complex as previously described (Dueber et al., 2009). Glucaric acid titer was found to be strongly affected by the number of scaffold interaction domains targeting upstream Ino1 enzymes, whereas the effect of increased numbers of MIOX-targeted domains was much less significant. We determined that the scaffolds directly increased the specific MIOX activity and that glucaric acid titers were strongly correlated with MIOX activity. Overall, we observed an approximately 5-fold improvement in product titers over the non-scaffolded control, and a 50% improvement over the previously reported highest titers. These results further validate the utility of these synthetic scaffolds as a tool for metabolic engineering.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/20117231/)

---

### 26.Tunable Signal Processing in Synthetic MAP Kinase Cascades

**Abstract** 

&emsp;&emsp;The flexibility of MAPK cascade responses enables regulation of a vast array of cell fate decisions, but elucidating the mechanisms underlying this plasticity is difficult in endogenous signaling networks. We constructed insulated mammalian MAPK cascades in yeast to explore how intrinsic and extrinsic perturbations affect the flexibility of these synthetic signaling modules. Contrary to biphasic dependence on scaffold concentration, we observe monotonic decreases in signal strength as scaffold concentration increases. We find that augmenting the concentration of sequential kinases can enhance ultrasensitivity and lower the activation threshold. Further, integrating negative regulation and concentration variation can decouple ultrasensitivity and threshold from the strength of the response. Computational analyses show that cascading can generate ultrasensitivity and that natural cascades with different kinase concentrations are innately biased toward their distinct activation profiles. This work demonstrates that tunable signal processing is inherent to minimal MAPK modules and elucidates principles for rational design of synthetic signaling systems.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/21215374/)

---

### 27.Production of the antimalarial drug precursor artemisinic acid in engineered yeast

**Abstract** 

&emsp;&emsp;Malaria is a global health problem that threatens 300-500 million people and kills more than one million people annually. Disease control is hampered by the occurrence of multi-drug-resistant strains of the malaria parasite Plasmodium falciparum. Synthetic antimalarial drugs and malarial vaccines are currently being developed, but their efficacy against malaria awaits rigorous clinical testing. Artemisinin, a sesquiterpene lactone endoperoxide extracted from Artemisia annua L (family Asteraceae; commonly known as sweet wormwood), is highly effective against multi-drug-resistant Plasmodium spp., but is in short supply and unaffordable to most malaria sufferers. Although total synthesis of artemisinin is difficult and costly, the semi-synthesis of artemisinin or any derivative from microbially sourced artemisinic acid, its immediate precursor, could be a cost-effective, environmentally friendly, high-quality and reliable source of artemisinin. Here we report the engineering of Saccharomyces cerevisiae to produce high titres (up to 100 mg l(-1)) of artemisinic acid using an engineered mevalonate pathway, amorphadiene synthase, and a novel cytochrome P450 monooxygenase (CYP71AV1) from A. annua that performs a three-step oxidation of amorpha-4,11-diene to artemisinic acid. The synthesized artemisinic acid is transported out and retained on the outside of the engineered yeast, meaning that a simple and inexpensive purification process can be used to obtain the desired product. Although the engineered yeast is already capable of producing artemisinic acid at a significantly higher specific productivity than A. annua, yield optimization and industrial scale-up will be required to raise artemisinic acid production to a level high enough to reduce artemisinin combination therapies to significantly below their current prices.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/16612385/)

---

### 28.Robust multicellular computing using genetically encoded NOT gates and chemical 'wires'

**Abstract** 

&emsp;&emsp;Computation underlies the organization of cells into higher-order structures, for example during development or the spatial association of bacteria in a biofilm. Each cell performs a simple computational operation, but when combined with cell-cell communication, intricate patterns emerge. Here we study this process by combining a simple genetic circuit with quorum sensing to produce more complex computations in space. We construct a simple NOR logic gate in Escherichia coli by arranging two tandem promoters that function as inputs to drive the transcription of a repressor. The repressor inactivates a promoter that serves as the output. Individual colonies of E. coli carry the same NOR gate, but the inputs and outputs are wired to different orthogonal quorum-sensing 'sender' and 'receiver' devices. The quorum molecules form the wires between gates. By arranging the colonies in different spatial configurations, all possible two-input gates are produced, including the difficult XOR and EQUALS functions. The response is strong and robust, with 5- to >300-fold changes between the 'on' and 'off' states. This work helps elucidate the design rules by which simple logic can be harnessed to produce diverse and complex calculations by rewiring communication between cells.

> [全文链接](https://pubmed.ncbi.nlm.nih.gov/21150903/)

---

### 29.Division of logic labour

**Abstract** 

&emsp;&emsp;Computation forms the foundation of electronic devices that pervade our daily lives. Many of these devices run on digital circuits assembled from logic gates. All logic gates use an unambiguous rule to convert inputs of 0 or 1 into outputs — again, of 0 or 1. Depending on their basic operation, these gates are given names such as AND, OR and NOR. Computation is also fundamental to numerous biological functions, from information processing by neural networks to nutrient sensing by microbes. In this issue, Tamsir et al.1 and Regot et al.2 describe an innovative multicellular strategy for engineering complex logic circuits, which can potentially respond to combinations of biological signals by generating a useful read-out.

> [全文链接](https://www.nature.com/articles/469171a)

---

### 

**Abstract** 

&emsp;&emsp;

> [全文链接]()

---

### 

**Abstract** 

&emsp;&emsp;

> [全文链接]()

---

### 

**Abstract** 

&emsp;&emsp;

> [全文链接]()

---