---
layout:     post
title:      2022-04-01-疼痛遗传学100篇paper
subtitle:   第一部分——第1到第10篇paper
date:       2022-04-01
author:     DL
header-img: img/home-bg-o.jpg
catalog: true
tags:
    - Pain
---

### 1.Title: "Sources of Individual Differences in Pain (2021)."

[Mogil, J. S. (2021). "Sources of Individual Differences in Pain." Annu Rev Neurosci 44: 1-25.](https://doi.org/10.1146/annurev-neuro-092820-105941)

&emsp;&emsp;Pain is an immense clinical and societal challenge, and the key to understanding and treating it is **variability**. Robust interindividual differencesare consistently observed in pain sensitivity, susceptibility to developingpainful disorders, and response to analgesic manipulations

**PAIN AS A CLINICAL AND SOCIETAL PROBLEM**

&emsp;&emsp;By a variety of metrics, pain is the leading source of morbidity worldwide.

&emsp;&emsp;I have long argued that **individual differences are the key to understanding chronic pain** (Mogil1999, 2012a; Mogil et al. 1996).

&emsp;&emsp;As it has been shown repeatedly that **high acute pain sensitivity** is a risk factor for developing chronic pain (see Nielsen et al. 2009), individual differences in acute (including experimental) pain are considered as well.

**INDIVIDUAL DIFFERENCES IN PAIN**

&emsp;&emsp;As pain is an utterly **subjective experience**, there is no way to ever know, although people do generally employ highly similar verbal descriptors after similar noxious insults (Melzack 1975). The individual differences of concern herein are notqualitative but rather **quantitative**, that is, continuous quantifiable variation in: pain threshold, pain intensity, pain tolerance, pain duration, pain endophenotypes (such as summation or condi-tioned pain modulation), susceptibility to developing pain from injury or pathology, and response to treatment. It should also be made clear that although variation can be formally divided into within-subject and **between-subject variability**, it is only the latter that is of interest here.

&emsp;&emsp;In fact, the work of Donald Price and colleagues (e.g., Price et al. 1983) has shown that the **pain rating scales** still in common use have **admirable validity and reliability** and represent true ratio scale measures for both experimental and clinical pain. Similarly, pain thresholds measured similarly on different days show very **high within-subject correlations** (e.g., Isselee et al. 1997). It has also been demonstrated that variability in pain ratings correlates well with variability in purported biomarkers such as functional magnetic resonance imaging of cortical (Coghill et al. 2003) and subcortical (Khan& Stroman 2015) regions. Thus, there is every reason to believe that **individual differences in subjective pain reports reflect true individual differences in pain perception**.

&emsp;&emsp;Some participants rated the stimulus as barely registering as pain, whereas for others it was at the level of their “worst pain imaginable.”

&emsp;&emsp;Variability in clinical pain intensity and susceptibility to **developing clinical pain** in the first place is also robust.

&emsp;&emsp;Given that much persistent **postoperative pain is neuropathic** in nature ( Johansen et al.2012), it might seem like much of this variability could be explained simply by whether the surgeon did or did not injure any nerves during the surgical procedure. 

&emsp;&emsp;Finally, large variability in the **efficacy of both pharmacological and nonpharmacological analgesic treatments** has been repeatedly noted.

&emsp;&emsp;Suffice it to say, whatever pain-related parameter is considered, the amount of variance encountered is extraordinarily high. What can account for this variability? The answer to that questionis the answer to all questions of variability: nature and nurture.

**GENETIC EXPLANATIONS FOR PAIN VARIABILITY**

&emsp;&emsp;The genetic revolution of the past few decades has centered much attention on the potential role of **genotype** in producing **individual differences in pain**. Christopher Nielsen et al. (2012) produced an excellent review of **56 twin studies** published up to 2012, providing evidence of heritability(i.e., the percentage of the overall trait variability accounted for by inherited genetic factors) of around **25–50%** for various pain syndromes. A 2013 review of 27 twin studies of low back painrevealed heritability ranging from 21% to 67% (Ferreira et al. 2013).

&emsp;&emsp;**Table 1 shows heritability estimates of published twin studies from 2013 to 2020, many of them performed on the large TwinsUK Registry, which also show a median heritability of 36%.**

[![bD4oIx.png](https://s1.ax1x.com/2022/03/06/bD4oIx.png)](https://imgtu.com/i/bD4oIx)

&emsp;&emsp;However, twin studies of **experimental pain in humans** fall into exactly the same range, from **10% to 60%** (MacGregor et al.1997, Nielsen et al. 2008, Norbury et al. 2007). Finally, the same range of heritability estimates can be gleaned from surveys of **inbred mouse strains**; the heritability of 22 nociceptive assays ranged from **30% to 76%**, with a median heritability of 46% (Lariviere et al. 2002, Mogil et al. 1999).

&emsp;&emsp;Until recently, most studies addressing the genetic mediation of pain and pain inhibition havebeen performed in mice and, to a lesser extent, rats.

&emsp;&emsp;Genetically variable responses to analgesics depend on the modalityof pain being inhibited by those analgesics.

&emsp;&emsp;**Over 20 quantitative trait loci (QTLs)** — that is, chromosomal regions containing gene variants affecting trait variability — have been identified (Liang et al. 2014; Recla et al. 2014, 2019; Sorge et al. 2013; Wieskopf et al. 2015; see also Mogil2012a). In some cases, the genes affected by these variants have never been identified; in others,the likely genes responsible are known. In a few cases, genetic association data (see below) or pharmacological data from **human cohorts** confirmed the relevance of the gene in our species (Mogilet al. 2003, 2011; Nissenbaum et al. 2010; Sorge et al. 2012; Wieskopf et al. 2015). As predictedby known genotype-sex interactions, a fair number of these QTLs were **sex dependent** (see Mogil2003). In addition to QTL mapping, **pain-related genes** have been identified in rodent models using microarray gene expression profiling (Costigan et al. 2010, Tegeder et al. 2006, Vicuna et al.2015) and candidate gene testing via transgenic null mutants** (see LaCroix-Fralish et al. 2007,Mogil & Grisel 1998).

&emsp;&emsp;Although genetic analyses of pain continue to be performed in rodents, the advent of **genetic association study** methodologies convinced most investigators to **switch to humans** as research subjects. Indeed, DNA sequence–based genetic techniques are among the few experimental approaches that can be performed as readily **in humans as in laboratory animals**. Broadly speaking, two types of human genetic studies are feasible: those **aimed at identifying the genes underlying highly penetrant (but rare) monogenic disorders, and genetic association studies using large cohorts**.

&emsp;&emsp;Of these two techniques, the former has clearly been more successful thus far. Genes whose loss-of-function variants cause known forms of congenital insensitivity (or indifference) to pain have been identified, as have genes whose gain-of-function variants cause painful disorders such as primary erythromelalgia, paroxysmal extreme pain disorder, and familial hemiplegic migraines. These advances have been reviewed in detail previously (e.g., Barrett et al. 2008, Foulkes & Wood2008). They will not be considered here, however, because although they are potentially important advances both for the sufferers of rare monogenic disorders and as potential targets for analgesic drug development, **there is very little evidence that they contribute to individual differences in pain sensitivity or chronic pain susceptibility in the population at large**.

&emsp;&emsp;To explain the genetic contribution to individual differences in humans, one must engage in **complex trait genetics via genetic association studies**. In general, these studies attempt to provide statistical evidence for the association between trait variance (either so-called caseness or severity) and **single nucleotide polymorphisms (SNPs**; also known as genetic variants or, when sufficiently rare, mutations) in or near potentially trait-relevant genes (see Lander & Schork 1994). The first wave of such studies included **single-gene (or in some cases, small panels of genes) association studies** in which the genes being examined were already known or suspected to be relevant to pain based on the known actions of the proteins they code for. As with the candidate gene association study technique more generally (Tabor et al. 2002), **candidate gene association studiesof pain have not proven to be broadly reproducible** (Mogil 2009). For example, de Vries et al.(2016) were unable to find evidence—in a much larger cohort than any of the original studies—of any of 27 published genetic associations with migraine. A similar exercise was performed by Klepstad et al. (2011), who failed to find evidence for association of any of 25 candidate genes with opioid analgesic efficacy in a sample of 2,294 cancer pain patients. A formal meta-analysis of the commonly studied 118A→G (N40D; rs1799971) variant of the OPRM1 gene encoding the μ-opioid receptor revealed no significant associations with pain and only a weak association with opioid requirements in clinical settings (Walter & Lötsch 2009). A meta-analysis of candidate gene association studies of neuropathic pain revealed no significant associations with variants of OPRM1 or COMT (catechol-O-methyltransferase) but a significant association with various HLA (humanlymphocyte antigen) alleles (Veluchamy et al. 2018).

&emsp;&emsp;**It is very likely that COMT variants do indeed contribute to pain variability, but only in some pain states and in ways that have yet to be clearly elucidated.**

&emsp;&emsp;In contrast to the poor reproducibility of candidate gene association studies, much more highly powered **genome-wide association studies (GWASs) (Altshuler et al. 2008) do tend to replicate well, especially given that modern GWASs generally feature independent replication samples within them**. Given the high cost of such endeavors, **only a limited number of pain-relevant GWASs have been published**. Very recently, the **statistical power** of these studies has been enhanced by the availability of extremely large genotyped cohorts in the public and private domain, although generally at the expense of any detailed pain-specific phenotypic information. **Of all pain syndromes, migraine is the most advanced in terms of genetic elucidation by GWAS**, with the most recent meta-analysis of 375,000 individuals (including almost 60,000 migraineurs) confirming 38 relevant genomic regions associated with migraine (Gormley et al. 2016). 

&emsp;&emsp;**Extant pain-related GWAS findings for pain states other than headache are detailed inTable 2.**The vast majority of the genes on the list do not code for well-known molecules in the pain research field. For example, only five genes listed in Table 2—NGF,ESR1,OPRM1,GFRA2,and LMX1B—have been studied for pain via the transgenic knockout mouse approach.

[![bDxRqP.png](https://s1.ax1x.com/2022/03/06/bDxRqP.png)](https://imgtu.com/i/bDxRqP)

[![bDxoGQ.png](https://s1.ax1x.com/2022/03/06/bDxoGQ.png)](https://imgtu.com/i/bDxoGQ)

[![bDxqrq.png](https://s1.ax1x.com/2022/03/06/bDxqrq.png)](https://imgtu.com/i/bDxqrq)

&emsp;&emsp;There are a number of ways one might explain this **lack of concordance**, including (a) potential species differences between mice and humans, (b) GWAS genes being relevant to developing the pathology itself but not to pain within that pathology, and (c) the possibility that our current list of molecular players in pain representsonly the tip of a much larger iceberg.

&emsp;&emsp;Also striking is the fact that, although most of the associations in Table 2 were replicated within the GWAS study reporting them, there are **only a limited number of between-study replications and virtually no examples (DCC,HLA) of a gene showing up as associated with different pain syndromes**. This latter fact suggests that different types of painful disorders have highly **heterogeneous genetic determinants**, which in turn suggests that the genes being identified are related more to susceptibility to developing the painful disorder than to pain levels within the disorder.

&emsp;&emsp;Although the replication problem has essentially been solved using GWAS methodology, especially in tandem with huge sample sizes, issues remain. **Most published GWASs do not feature any functional follow-up studies**, such that the causal gene cannot be unambiguously determined from the index (most highly associated) SNP. Local genomic effects are turning out to be much morecomplex than we envisaged. For example, the intronic SNP rs9349379, which is associated with migraine in multiple GWASs, was linked to PHACTR1, but detailed follow-up studies revealed that the SNP actually influenced the expression of EDN1, more than 600,000 base pairs upstream (Gupta et al. 2017). The main and most-discussed limitation concerns the **small sizes of the genetic effects** themselves.

&emsp;&emsp;It is now becoming apparent that the total number of genes contributing to heritable traits is much larger than we had previously conceived; complex traits are not polygenic but rather omnigenic, mediated by huge numbers (e.g.,>100,000for height) of causal variants (Boyle et al. 2017). In other words, a large fraction of all the genes there are contribute to variation in disease risk, and pain appears to be no exception.

**SEX DIFFERENCES IN PAIN AS A SOURCE OF VARIABILITY**

&emsp;&emsp;**Women are more sensitive to pain than men,regardless of the noxious stimulus modality or pain measure.** A more recent review (Mogil 2020) of the laboratory animal literature on the topic provides convergent evidence: 85% of published findings in rodents show females to be more sensitive to pain than males.

&emsp;&emsp;More importantly, the biological underpinnings of pain in males and females are proving to be robustly different in terms of underlying genetic, neurochemical, and especially neuroimmune components. The relevant question here is to what extent such sex differences drivevariability in pain response. In mice, this question has been addressed explicitly. In an analysis of all sources of variation in more than 8,000 baseline measurements of acute, thermal pain in mice, **sex ranked only sixth of eight factors identified via a machine learning algorithm, behind experimenter, genotype, season, cage density, and time of day.**

&emsp;&emsp;Human experiments explicitly studying how much variability in pain sensitivity is explained by sex/gender are rare, even as many have observed statistically significant sex differences. In a study of multiple experimental pain modalities in human volunteers, Doehring et al. (2011) observed that** sex accounted for 1–15% of the variance (depending on modality)**.

&emsp;&emsp;Related to the topic of sex differences and also likely contributing to overall variability are the within-female states of **pregnancy and hormonal cyclicity**. 

**OTHER ORGANISMIC FACTORS AND PAIN VARIABILITY**

&emsp;&emsp;Other factors intrinsic to the organism, apart from genotype and sex, account for variability in pain, and the most important of these is probably **age**. Age is, of course, the biggest or among the biggest risk factors for certain common pain conditions, including osteoarthritis (Felson et al.2000) and postherpetic neuralgia (Forbes et al. 2016). A recent meta-analysis concluded that older participants display significantly increased sensitivity (lower thresholds) to **pressure pain** than younger adults but no differences in **heat pain** sensitivity (El Tumi et al. 2017). A common finding in this literature is that young children appear to be more sensitive to noxious stimuli than older children (El Tumi et al. 2017).

&emsp;&emsp;One organismic factor that has received renewed attention in recent years is **racial/ethnic differences**. Racial/ethnic minorities appear to have higher frequencies and intensities of clinical painand to be more adversely affected by chronic pain (e.g., lower quality of life, higher disability), but it is unclear whether this represents a racial/ethnic difference in pain sensitivity per se and/or is due to disparities in access to and quality of care (see Kim et al. 2017). Kim et al. (2017) performed a systematic review and meta-analysis of 41 studies examining racial/ethnic differences (among non-Hispanic whites, Blacks, Asians, and Hispanics) in laboratory-controlled tests of experimental pain. Overall, **minority groups demonstrated significantly higher pain ratings and lower pain tolerance; there were no differences in pain thresholds**. This conclusion is similar to that of a prior quantitative review (Rahim-Williams et al. 2012). However, heterogeneity across studies in this subfield is high, effect sizes are small, confounds remain (especially with respect to immigration generational status), and it remains entirely unclear whether the differences seen can be attributed to race or ethnicity. Apart from race, other visible phenotypic features that have been proposed tobe associated with **experimental pain sensitivity** include eye and hair color, with those with **darker pigmentation** displaying higher sensitivity (Holmgaard et al. 2017), and **body weight** (Tashani et al.2017, VanDenKerkhof et al. 2011), specifically in areas with excess subcutaneous fat (Price et al.2013).

&emsp;&emsp;**Personality traits** are organismic factors that have been posited as important sources of variability in pain. Although there have been many studies asserting associations of personality dimensions(e.g., neuroticism, harm avoidance, novelty seeking) on standardized personality questionnaires with pain (e.g., Timmerans & Sternbach 1974), much more attention has in fact been paid to more specific psychological constructs. A very recent umbrella review of systematic reviews of **psychological risk factors** for musculoskeletal pain (Martinez-Calderon et al. 2020) identified the following as involved in onset and/or persistence (in alphabetical order): anger/frustration, anxiety, cognitive avoidance, coping, depression, expectations of recovery, external locus of control,fear avoidance, illness behaviors, kinesiophobia, mental health, negative affect, pain catastrophizing, perceived health, perceived stress, poor sleep patterns, psychological distress, self-efficacy,somatization, type A behavior, and unhelpful beliefs.

&emsp;&emsp;Of these psychological traits (and/or, potentially, states), **anxiety and depression** are the most well studied (see Bair et al. 2003, Gureje 2008) (including in animal models; see Liu & Chen 2014), although it remains unclear whether they are risk factors for pain, effects of having pain, and/or comorbid conditions. 

**WHAT ACCOUNTS FOR PAIN VARIABILITY?**

&emsp;&emsp;It nonetheless remains true that explaining individual differences in pain remains a critically important task in pain research. Any true implementation of personalized medicine for pain will require such knowledge, and it is likely that molecular and anatomical elements contributing to pain variability would make particularly tempting targets for analgesic intervention.

---

### 2.Title: "Individual Differences in Pain: Understanding the Mosaic that Makes Pain Personal (2017)."

[Fillingim, R. B. (2017). "Individual differences in pain: understanding the mosaic that makes pain personal." Pain 158 Suppl 1: S11-S18.](https://journals.lww.com/pain/Fulltext/2017/04001/Individual_differences_in_pain__understanding_the.3.aspx)

&emsp;&emsp;The experience of pain is characterized by **tremendous inter-individual variability**. Multiple biological and psychosocial variables contribute to these individual differences in pain, including demographic variables, genetic factors, and psychosocial processes. For example, **sex, age and ethnic group differences** in the prevalence of chronic pain conditions have been widely reported. Moreover, these **demographic factors** have been associated with responses to experimentally-induced pain. Similarly, both **genetic and psychosocial factors** contribute to clinical and experimental pain responses. Importantly, **these different biopsychosocial influences interact with each other in complex ways to sculpt the experience of pain**. Some genetic associations with pain have been found to vary across sex and ethnic group. Moreover, **genetic factors** also interact with psychosocial factors, including stress and pain catastrophizing, to influence pain. The individual and combined influences of these biological and psychosocial variables results in a unique mosaic of factors that contributes pain in each individual. Understanding these mosaics is critically important in order to provide optimal pain treatment, and future research to further elucidate the nature of these biopsychosocial interactions is needed in order to provide more informed and personalized pain care.

&emsp;&emsp;It has long been appreciated that individuals differ from each other in important ways. More than 2,000 years ago Plato said: “No two persons are born exactly alike; but each differs from the other in natural endowments (360 B.C.).”

&emsp;&emsp;**The genomic revolution and the ensuing promise of precision medicine have reinvigorated and legitimized scientific interest in individual differences.**

&emsp;&emsp;First, I will **introduce the topic** of individual differences in responses to pain and its treatment, including a biopsychosocial context for conceptualizing individual differences. Then, I will **present findings** regarding demographic factors that are associated with individual differences in pain. Next, I will **discuss genetic and psychosocial contributions** to individual differences, and I will **present examples of interactions** among these multiple individual difference factors. I will describe the **clinical implications** of individual differences in pain, followed by conclusions and recommendations for future research.

&emsp;&emsp;By **definition pain** is a subjective and highly personal experience, which presents challenges for both the researcher and clinician. A well-recognized challenge resulting from the subjective nature of pain is that direct measurement of pain is impossible, rather we must rely on **individuals’ self-report**, and to some extent their behavior, to provide a glimpse into their experience. However, an equally important but less often discussed challenge results from the **highly personal nature of the pain experience**; the experience of pain is **sculpted by a mosaic of factors** unique to the person, which renders the pain experience completely individualized. That is, there are pervasive and important individual differences in pain, and these individual differences produce pain experiences that are completely **unique to the person experiencing them** (i.e. they make the pain personal).

&emsp;&emsp;**Define individual differences in pain** as between person differences in the pain experience that are independent of the initiating stimulus. Perhaps the simplest manifestation of individual differences is that an **experimental stimulus** delivered at a standardized intensity elicits subjective pain reports that vary dramatically between individuals. Such individual differences also emerge in the **clinical environment**. For example, pain reports following the same surgical procedure vary greatly across patients. Similarly, responses to pain treatments are characterized by robust individual differences.

&emsp;&emsp;The **biopsychosocial model** provides an ideal framework for conceptualizing individual differences in pain. This model posits that the experience of pain is influenced by **complex and dynamic interactions** among multiple biological, psychological, and social factors. Importantly, the ensemble of biopsychosocial factors contributing to the experience of pain and its expression varies considerably across people. Thus, pain is sculpted by a mosaic of factors that is completely unique to each individual at a given point in time, and this mosaic must be considered in order to provide optimal pain treatment.

&emsp;&emsp;Notably, some markers may reflect **mechanisms underlying pain**, while others do not. Examples of the former include demographic factors, such as **sex, race/ethnicity and age**. While each of these variables has been associated with pain responses (as discussed below), they **reflect proxies for mechanisms influencing pain** rather than mechanisms themselves. That is, the sex of an individual does not directly influence pain, rather sex differences in pain reflect the effects of other biological and psychosocial processes (e.g. sex hormones, inflammatory responses, gender roles, pain coping).

&emsp;&emsp;Indeed, investigating the factors contributing to individual differences in pain can provide important insights into **pain mechanisms**, which may lead to the development of novel treatments. Also, incorporating an understanding of individual differences into assessment and diagnosis of pain in the **clinical setting** may allow the clinician to select treatments that are tailored to the patient, thereby improving treatment outcomes.

**Sex Differences**

**Race/Ethnic Group Differences**

&emsp;&emsp;The concepts of race and ethnicity are complex biological and social constructs that **remain poorly defined**. Nahin found that pain prevalence was **lowest among Asians** compared to other race/ethnic groups in the US. Other studies of adults in the **US have reported higher prevalence of persistent pain** among whites compared to other racial/ethnic groups. Among older adults some studies have reported **higher pain prevalence among minorities** compared to whites, while others reported no differences in pain prevalence. While there is conflicting information regarding pain prevalence may be lower among minority versus majority ethnic groups, studies consistently suggest that the severity and impact of **pain appears to be greater among minorities** who are experiencing chronic pain. Indeed, our own studies demonstrate greater pain severity and functional limitations among **African Americans** compared to non-Hispanic whites with knee osteoarthritis. In addition, **differences in pain perception** between racial/ethnic groups may contribute to differences in **severity of clinical pain**. A meta-analytic review of studies examining pain perception in generally healthy adults found that **African Americans display greater experimental pain sensitivity** compared to non-Hispanic whites. Similarly, our recent findings among adults with knee osteoarthritis showed greater pain sensitivity and temporal summation of pain among **African Americans**. These findings are largely based on work conducted in the United States, where racial and ethnic disparities in health are a substantial national concern. Similar findings have emerged in other developed countries throughout the world; however, little data related to ethnic group differences in pain have been reported from less developed countries.

&emsp;&emsp;The mechanisms underlying racial/ethnic group differences in the experience of pain are inevitably **multifactorial, and include factors related to socioeconomic standing and access to adequate health care**. For example, in most developed countries, members of minority groups on average have lower socioeconomic status, which has been associated with increased pain prevalence and more severe pain. In addition, considerable evidence suggests that minority patients are at greater risk for **undertreatment of their pain**, which could obviously contribute to the greater clinical pain severity observed among members of minority groups. **Pain coping** also differs significantly across racial/ethnic groups, and it is possible that biological factors, such as **genetic contributions**, may play a role in racial/ethnic differences in pain responses.

**Age-Related Differences**

**Interactions Among Biopsychosocial Factors**

---

### 3.Title: "Sodium-calcium exchanger-3 regulates pain ‘wind-up’: from human psychophysics to spinal mechanisms."

**Abstract**

&emsp;&emsp;Repeated application of noxious stimuli leads to a progressively increased pain perception; this temporal summation is enhanced in and predictive of clinical pain disorders. Its electrophysiological correlate is “wind-up”, in which dorsal horn spinal neurons increase their response to repeated nociceptor stimulation. To understand the genetic basis of temporal summation we undertook a GWAS of wind-up in healthy human volunteers and found significant association with SLC8A3 encoding Sodium-Calcium exchanger type-3 (NCX3).  Our study highlights Ca2+efflux as a pathway underlying temporal summation and persistent pain which may be amenable to therapeutic targeting.

**Cover letter**

&emsp;&emsp;**Chronic pain** affects 1 in 5 of the general population and current treatments are inadequate. The **spinal cord** is known to be a locus for **central sensitisation** leading to chronic pain and in particular wind-up (the progressive increase in the response of dorsal horn neurons due to repetitive stimulation of C-fibre nociceptors), which leads to temporal summation of pain and central sensitisation. This **activity-dependent plasticity** within the spinal dorsal horn is a critical pathophysiological mechanism underlying chronic pain states. We still have an incomplete understanding of how this plasticity arises, especially in humans. Here we applied a genetic approach to the characterization of the molecular basis of wind-up. We first carried out a **genome-wide association study (GWAS)** in a human cohort characterised for an experimental model of temporal summation of pain. We then validated the **top candidate gene** (NCX3), identified in the GWAS, using mouse models. Our findings highlight the role of Ca2+efflux in determining spinal plasticity and we show that this is a **novel target for the treatment** of persistent pain states.

&emsp;&emsp;Secondly, **no group has ever previously shown a genome-wide significant association in an experimental pain model**, probably as a consequence of the complexity of validating and then capturing complex pain phenotypes. We have achieved this through carefully **controlled human experimental conditions**. We identified a very strong candidate in NCX3, a cation exchanger widely expressed in the CNS. We have then undertaken detailed physiological analyses in the mouse and have shown that NCX3 mediates exactly the same phenotype of temporal summation. We also delineated the molecular and electrophysiological mechanism. Our study therefore illustrates the power of **integrating human and mouse genetic approaches in understanding the nervous system**. Finally, pain remains a large unmet need as exemplified by the current opioid crisis. We desperately need **new non-addictive pain therapies**. Pharmaceutical companies are increasingly orienting analgesic drug development to targets with human genetic evidence (as exemplified here). Not only have we identified a novel target,but we have also demonstrated efficacy in targeting this pathway by reducing central sensitisation at the spinal level. 

**Introduction**

&emsp;&emsp;In humans, repetitive or prolonged noxious stimulation results in increased pain perception, a psychophysical phenomenon termed **temporal summation**. This is frequency dependent, occurs following application of high threshold mechanical, electrical or thermal stimuli  and can be elicited from skin, musculoskeletal and visceral tissues. Electrophysiological assessment of nociceptor activity and also the withdrawal reflex in humans have shown that such temporal summation is due to activity-dependent plasticity within the central nervous system (CNS). An important neural correlate of this temporal summation of pain is the phenomenon of ‘**wind-up**’ of the responses of dorsal horn (DH) neurons within the spinal cord. **Wind-up** refers to the phenomenon that repetitive stimulation of C-fibre nociceptors evokes a progressively increased response in predominantly deep DH neurons over the course of the stimulus train; this is frequency-dependent, occurring when stimuli are delivered between 0.5 and 5Hz and requires C-fibre activation. Wind-up has been reported in **rodent, cat and primate DH** and is one of the neural mechanisms underlying central sensitisation (the increased responsiveness of nociceptive neurons in the CNS to their normal or subthreshold afferent input).

&emsp;&emsp;The recent standardisation of **quantitative sensory testing protocols** to assess the psychophysical response to sensory stimuli in humans and their application to **larger cohorts **is helping to inform our understanding of the **inter-individual variation in pain perception**. **Twin studies have shown a significant heritability** of these evoked pain phenotypes (up to 50%), although the degree of such heritability depends on the modality. Determining the genetic basis of these pain phenotypes in humans could provide fundamental understanding of the **neurobiology of pain** and potentially also identify **novel treatment targets**. A number of these experimental pain measures including temporal summation have been shown to be **enhanced in clinical disorders** including musculoskeletal, visceral and neuropathic chronic pain. Furthermore, temporal summation has been reported to be **predictive of clinical disorders** such as post-surgical pain. There have been a number of **candidate gene studies** applied to quantitative sensory testing including temporal summation, but many of these findings have yet to be replicated. Furthermore, there have been **no genome-wide association studies** (GWAS) which have reported significant associations with experimental pain traits. Here, we performed a GWAS for temporal summation of pain (expressed as wind-up ratio, WUR) in a sample of ~1,000 healthy volunteers of mixed European/Native American/African ancestry. WUR demonstrated genome-wide significant association with the **SLC8A3 locus encoding the sodium-calcium exchanger type 3 (NCX3)** which we then validated in mouse models demonstrating that this is a critical determinant of calcium handling and the activity-dependent plasticity of DH neurons. 

**Results**

&emsp;&emsp;**We obtained the wind-up ratio (WUR)as a measure of temporal summation in 1,061 individuals, recruited in Medellín (Colombia), with a median age of 23 (range 18-45) years, of which 55% were female.** WUR was evaluated over the mid ventral forearm (side randomised) using a 255mN von Frey hair (Touch Test, North Coast, USA). The pain rating (0-100) reported for a train of 10 stimuli (at 1 Hz) was divided by the pain rating of a prior (single) stimulus, and the mean of 5 repetitions was recorded as the WUR (Figure 1A). After quality control (see methods), the **median pain rating** for the single stimulus was 2.85 (IQR 3.85) and 4.00 (IQR 5.14) for the train of stimuli (Figure 1B, this increase was highly statistically significant, Wilcoxon signed rank P-value< 1E-16). The **median WUR** was 1.27 (IQR 0.46), with the trait distribution having a longer tail towards high values (Figure 1C). 

&emsp;&emsp;Individuals were **genotyped** on Illumina’s Omni Express chip containing 730,525 SNPs across the genome, of which 673,034 SNPs were retained after applying quality control filters (see methods). We performed **genotype imputation** based on 1000 Genomes data resulting in a final genetic dataset including 9,616,058 autosomal and X chromosome SNPs. After **quality control of the data for individuals**, 991subjects were retained for further analyses. 

&emsp;&emsp;Based on the genome-wide SNP data, the **average ancestry** of these individuals was estimated as: 29% Native American, 61% European and 10% African. Women were observed to  have  a  small  but  significantly  greater  WUR,  relative  to  men  (1.51  vs. 1.40; Spearman’s P-value 1.3×10-7).  Native  American  ancestry  and  self-reported  depression  score  had  a  low  but  significant **correlation with WUR** (Pearson’s r = 0.06, P-value 1×10-2, and r =0.05, P-value2×10-2, respectively). No significant effect for age was observed. 

&emsp;&emsp;**Association testing** detected three clusters of SNPs, in 1q21.3, 14q24.2, and 14q21.2, exceeding the threshold for genome-wide significance (P-value < 5.3×10-8; Figure 1). The SNPs in 1q21.3 are intergenic, while the SNPs in 14q24.2 overlap the SCL8A3/NCX3 gene (encoding the Na+/Ca2+exchanger 3),and those in 14q21.2 overlap LINC00871(long intergenic non-protein coding RNA 871). These SNPs have a **low polymorphism** in Europeans and Native Americans, with **minor allele frequencies** >10% being observed only in Africans, and explain 3-5% of the **phenotypic variance**. Consistent with their admixed ancestry, **minor allele frequencies** at these SNPs in the Colombians examined are intermediate between Europeans/Native Americans and Africans.

&emsp;&emsp;We evaluated **replication of the association results** observed in the Colombians by examining data from the **OPPERA study cohort**, which has also been characterized for a temporal summation phenotype. Considering that the OPPERA cohort is ethnically quite diverse and substantially different in composition that the main cohort, we conducted a gene-level association test for replication, and found that it was strongly significant for NCX3 (P-value 3×10−8). 

&emsp;&emsp;**ENCODE RNAseq data** indicates maximal expression of NCX3 in the spinal cord, while **GTex data** shows that NCX3 is widely expressed in the human CNS. The associated SNPs in 1q21.3 are intergenic but closest to S100A16  encoding an EF-hand containing calcium binding protein. These SNPs have been identified as eQTLs for S100A16 in the GTex data. The GTex data also shows that S100A16 is expressed in human CNS although with relativelylow expression in mouse DH. The gene-level association did not,however,show significant replication for S100A16  in the OPPERA cohort.

**Discussion**

&emsp;&emsp;Undertaking an experimental pain model in humans, we have found a novel genome-wide association between variants in the SLC8A3 locus (encoding NCX3) and WUR (a measure of temporal summation of pain). NCX3 was shown to be highly expressed by DH neurons and whilst genetic ablation of NCX3 in the mouse did not alter acute pain thresholds, it did result in enhanced second phase responses of the formalin test which partly reflects central sensitisation. NCX3 was shown to regulate Ca2+ dynamics in DH neurons and in the absence of NCX3 electrophysiological wind-up was enhanced and could be achieved at a low frequency providing a molecular and electrophysiological correlate of the behavioural observations. Viral overexpression of NCX3 in the dorsal horn led to reduced pain behaviour in the second phase of formalin and so enhanced NCX3 activity may have therapeutic potential.

&emsp;&emsp;**Human experimental pain models** provide the opportunity to study pain perception using stimuli which are carefully controlled in terms of modality, location, intensity, and temporal profile. There have been several candidate gene studies but relatively few GWAS studies using such experimental pain models, likely reflecting the challenges of capturing this complex phenotype in sufficient participant numbers. One study evaluated association of SNPs (revealed by exome sequencing) in a subset of twins with extremes of **heat pain sensitivity**; although the threshold for genome-wide significance was not reached, pathway analysis revealed significant enrichment for variants in genes of the angiotensin pathway. We used a validated protocol for WUR as a means to study temporal pain summation in young healthy participants to mitigate confounders such as chronic disease and aging. A punctate mechanical stimulus was used (a von Frey hair) and so the pain intensity ratings were expectedly low in our healthy participants. However, repetitive stimulation showed highly significant temporal summation and the sensory phenotype did not serve as a clinical outcome, but rather to identify biologically relevant gene associations. Indeed, we found three genome regions which exceeded the genome-wide significance threshold for association with WUR in the sample studied here. The evidence is most compelling for **14q24.2**, where a number of associated SNPs overlap the **NCX3 gene**,and this was the focus of our subsequent functional studies in the mouse. The single SNP showing association on **1q21.3** is intergenic however it is in proximity to (and an eQTL for) the gene S100A16, again highlighting calcium homeostasis as this gene encodes an EF-hand containing calcium binding protein. NCX3 is highly expressed in the CNS, and consistent with this, recent GWAS have associated variants in NCX3 with anhedonia and insomnia (with suggestive association having been also found for bipolar disorder. Since SNPs in **14q24.2** associated with WUR are non-coding, their phenotypic impact is most likely mediated through regulation of gene expression in the region, possibly of NCX3. The SNPs showing significant association in the Colombian sample overlap introns 2-6 of NCX3, a region that includes several enhancer-like signatures in the neural progenitor cells analysed by the **ENCODE consortium**. Furthermore, **GTex data** indicates that several SNPs in the NCX3 region represent eQTLs impacting on the expression of NCX3 in the brain. Although no such evidence is available for the SNPs associated here  with  WUR,  this  can  be  explained  by  these  SNPs  being  monomorphic  in  **Europeans** and the current paucity of gene expression data in **African ancestry populations**. It is very relevant to note here that most of the current GWAS participants are of **European ethnicity**, and therefore the studies are missing out on the huge genetic diversity in the rest of the world. Studies of ethnically diverse populations are crucial for a broader understanding of the biological basis of many traits;e.g. by working with the admixed Latin American populations, the CANDELA consortium was able to discover the contributions to multiple aspects of human physical variation by SNPs that are unique to Native American and East Asian ethnicities.

&emsp;&emsp;**In conclusion, our data from a human experimental pain GWAS identified association between the NCX3 locus and temporal summation of pain**. We then provided direct validation of NCX3 as a gene which regulates wind-up and central sensitisation within the spinal DH of mice. Our findings in which spinal overexpression of NCX3 could reduce pain behaviour suggest that selective activators of NCX3 would reduce temporal summation of pain in pathological states. Selective small molecule activators of NCX3 have not yet been developed, however targeting of NCX isoform-specific calcium-binding domains may provide a route to such selectivity.

---

### 4.Title: "Single-cell transcriptomic analysis of somatosensory neurons uncovers temporal development of neuropathic pain (2021)."

[Wang, K., et al. (2021). "Single-cell transcriptomic analysis of somatosensory neurons uncovers temporal development of neuropathic pain." Cell Res 31(8): 904-918.](https://www.nature.com/articles/s41422-021-00479-9)

**Introduction**

&emsp;&emsp;Nerve injury could cause neuropathic pain, which affects 7%–10% of the general population. Neuropathic pain severely debilitates patients’ life quality.

&emsp;&emsp;Taken together, the mechanisms underlying neuropathic pain are intricate. As a result, neuropathic pain is still hard to cure. **Clinically, the drugs for treating neuropathic pain are usually addictive or less potent.** In addition, some may evoke considerable side effects. Therefore, it is necessary to further explore the underlying mechanism for the occurrence and development of neuropathic pain at molecular and cellular levels.

---

### 5.Title: "Decomposition of pain into celltypes (2017)."

[Decomposition of pain into celltypes](https://cordis.europa.eu/project/id/740491)

&emsp;&emsp;**Almost 20% of the population has an ongoing pain problem. Pain is caused by a complex recruitment of different types of sensory neurons with different response-profiles and hence, the integrated response of an assembly of different neuronal types results in pain.**

&emsp;&emsp;Overall, this proposal takes a new system-wide strategy in that will affect development of new pain managing drugs, a field that has made little clinical advance the past century.

&emsp;&emsp;The importance of understanding the cellular and molecular mechanisms of chronic pain cannot be overestimated. **If we understand how chronic pain arises and is maintained, it will open for rational analgesic strategies that can make a difference for the nearly 8% of the population which have so severe chronic pain that it negatively affects the daily life. The overall objective of this project is to better understand the cellular and molecular causes for chronic pain.**

&emsp;&emsp;A general feature from our results is that **each pain sensitive neuron type expresses channels and receptors predicting a unique response profile, often to multiple stimulus modalities such as chemical, thermal, and mechanical. **Consistently, the different neuron types have variable contribution to the distinct sensory modalities and to chronic pain in animal models. As most of this work is performed in mouse as model organism, we have also determined the cellular basis for pain in non-human primate. This will allow the transfer of knowledge from research studies in the laboratory to humans in the clinic. The characterization of non-human primate neurons also allowed us to discover the neuronal type contributing to heritability to chronic pain in humans, which shows that pain is not one disorder, but several with different underlying molecular mechanisms.

&emsp;&emsp;Pain has historically been thought to arise from thin nerve endings located in the skin and tissues and organs of our body. We have discovered that the special cells, called **terminal Schwann cells, that surround the pain-sensing nerve cells, appear to be involved in sensing pain and that these cells forms a previously unknown sensory organ in the skin.** These terminal Schwann cells have an octopus-like shape with extensions that wrap around the ends of pain-sensing nerve cells that extend up into the outer layer of the skin. When artificially activating these cells, animal responses to heat, cold and pinprick pain is markedly increased. When artificially deactivating these cells, animals have a marked reduction in response to pinprick and pressure pain. Thus, we have discovered that pain-sensitive nerve cell terminals are not in fact always directly activated by a painful stimulus, but instead can be driven by associated terminal Schwann cells that build a web-like structure in the skin that works as a pain sensing organ. We have and will continue dissemination by means of publications in scientific journals, by attending and presenting at research meetings and by press releases and interviews with journalists.

---

### 6.Title: "Specialized cutaneous Schwann cells initiate pain sensation (2019)."

[Bullock, D., et al. (2019). "Reading and writing: the evolution of molecular pain genetics." Pain 160(10): 2177-2185.](https://www.science.org/doi/10.1126/science.aax6452)

**Abstract**

&emsp;&emsp;**An essential prerequisite for the survival of an organism is the ability to detect and respond to aversive stimuli.** Current belief is that noxious stimuli directly activate nociceptive sensory nerve endings in the skin. We discovered a specialized cutaneous glial cell type with extensive processes forming a mesh-like network in the subepidermal border of the skin that conveys noxious thermal and mechanical sensitivity. We demonstrate a direct excitatory functional connection to sensory neurons and provide evidence of a previously unknown organ that has an essential physiological role in sensing noxious stimuli. Thus, these glial cells, which are intimately associated with unmyelinated nociceptive nerves, are inherently mechanosensitive and transmit nociceptive information to the nerve.

**Introduction**

&emsp;&emsp;**The ability to detect and protect from damage-causing (noxious) stimuli relies on the existence of sensory afferents, called nociceptors.** Nociceptive nerves are generally unmyelinated and associate with Remak glial cells that protect and metabolically support the axons. **The unmyelinated nerve endings are activated by noxious stimuli and, hence, represent the pain receptors in the skin.**

---

### 7.Title: "Single cell transcriptomics of primate sensory neurons identifies cell types associated with chronic pain (2021)."

[Kupari, J., et al. (2021). "Single cell transcriptomics of primate sensory neurons identifies cell types associated with chronic pain." Nat Commun 12(1): 1510.](https://www.nature.com/articles/s41467-021-21725-z)

**Abstract**

&emsp;&emsp;**Distinct types of dorsal root ganglion sensory neurons may have unique contributions to chronic pain.** Identification of primate sensory neuron types is critical for understanding the cellular origin and heritability of chronic pain. 

---

### 8.Title: "Specialized cutaneous Schwann cells initiate pain sensation (2019)."

[Abdo, H., et al. (2019). "Specialized cutaneous Schwann cells initiate pain sensation." Science 365(6454): 695-699.](https://www.science.org/doi/full/10.1126/science.aax6452?casa_token=LzE_Gn06xkIAAAAA%3Aa50n2R0rslMweyU-T8RsW73Iee6_MRUmGqYyltXBkx2pWfONar3zWWO9YPGOY7NO0y7jDgEG6nX1n9h3)

**Abstract**

&emsp;&emsp;**An essential prerequisite for the survival of an organism is the ability to detect and respond to aversive stimuli. Current belief is that noxious stimuli directly activate nociceptive sensory nerve endings in the skin.**

**Introduction**

&emsp;&emsp;The ability to detect and protect from damage-causing (noxious) stimuli relies on the existence of sensory afferents, called nociceptors. Nociceptive nerves are generally unmyelinated and associate with Remak glial cells that protect and metabolically support theaxons. The unmyelinated nerve endings are activated by noxious stimuli and, hence, represent the pain receptors in the skin.

---

### 9.Title: "Genome-wide association reveals contribution of MRAS to painful temporomandibular disorder in males"

[Smith, S. B., et al. (2019). "Genome-wide association reveals contribution of MRAS to painful temporomandibular disorder in males." Pain 160(3): 579-591.](https://journals.lww.com/pain/Fulltext/2019/03000/Genome_wide_association_reveals_contribution_of.6.aspx)

**1.Introduction**

&emsp;&emsp;**To date, only one genome-wide association study (GWAS) of orofacial pain has been published.** Using self-reported orofacial pain as the phenotype, it found 2 female-specific genome-wide significant associations that were subsequently replicated in independent cohorts. One SNP was near the gene encoding relaxin/insulin-like family peptide receptor 2, RXP2; the second SNP was in the BAH domain and coiled-coil–containing gene, BAHCC1.

&emsp;&emsp;The aims of this study were to **undertake the first discovery GWAS of examiner-verified** chronic TMD, evaluate generalizability of the findings through independent replication, and investigate the functional biology of genetic markers that emerged from discovery and replication phases. The discovery sample included 3104 racially diverse participants from the Orofacial Pain: Prospective Evaluation and Risk Assessment (OPPERA) study, where **painful TMD was determined by clinical examination using modified Research Diagnostic Criteria for TMD (RDC/TMD), and genotyping was conducted using the Illumina HumanOmni 2.5M bead chip**. **Single-nucleotide polymorphisms that achieved genome-wide significance (P < 5 × 10−8) in the discovery cohort were carried forward** for replication in 7 additional cohorts.

**Methods 2.1 Discovery genome-wide association study cohort**

&emsp;&emsp;The discovery case–control cohort included 3104 subjects selected for genome-wide genotyping, drawn from the OPPERA study (n = 2868) and a separate case–control study of chronic TMD (n = 236). The OPPERA subjects were men and women (61.9% women) between the ages of 18 to 44 years, recruited at 4 sites in the eastern United States (Buffalo, NY; Baltimore, MD; Chapel Hill, NC; and Gainesville, FL); enrollment of this cohort has been described previously. The OPPERA subjects were racially and ethnically diverse: 57.5% white, 25.7% black, and 16.8% other/not reported.

**Methods 2.6. Genetic association analysis**

&emsp;&emsp;Genome-wide association was tested using PLINK 1.07 software,32 on discrete genotype and (for imputed SNPs) allele dosage data sets. Logistic regression models were constructed using additive SNP effects and included sex, age, race, and enrollment site. To account for population stratification, genetic ancestry was modeled using the first 3 principal components of the genotypes. The threshold for genome-wide significance was P < 5.0 × 10−8.

**Methods 2.7. Estimation of heritability**

&emsp;&emsp;Heritability of susceptibility to TMD was estimated using genome-wide SNP coverage using the GREML method implemented in the **software package Genome-Wide Complex Trait Analysis (GCTA).** This method estimates the contribution of additive SNP effects to observed variance to estimate narrow-sense heritability or the proportion of phenotypic variance attributable to additive allelic factors.

**Results 3.1. Discovery genome-wide association scan**

&emsp;&emsp;After quality control exclusions and genome-wide imputation, 999 painful TMD cases and 2031 pain-free controls were analyzed for association with ∼9.7M SNPs, **using logistic regression under an additive model** in both the full cohort and in females and males separately. The GREML method implemented in the software package **GCTA was used to estimate the variance explained by autosomal SNPs to estimate the narrow-sense heritability of chronic TMD**. This analysis indicated that about 17% of the phenotypic variance was accounted for by additive genetic factors.

---

### 10.Title: "Genome-wide association analysis of pain severity in dysmenorrhea identifies association at chromosome 1p13.2, near the nerve growth factor locus."

[Jones, A. V., et al. (2016). "Genome-wide association analysis of pain severity in dysmenorrhea identifies association at chromosome 1p13.2, near the nerve growth factor locus." Pain 157(11): 2571-2581.](https://journals.lww.com/pain/Fulltext/2016/11000/Genome_wide_association_analysis_of_pain_severity.23.aspx)

&emsp;&emsp;Dysmenorrhea is a common chronic pelvic pain syndrome affecting women of childbearing potential. Family studies suggest that genetic background influences the severity of dysmenorrhea, but genetic predisposition and molecular mechanisms underlying dysmenorrhea are not understood. In this study, we conduct the first genome-wide association study to identify genetic factors associated with dysmenorrhea pain severity. A cohort of females of European descent (n = 11,891) aged 18 to 45 years rated their average dysmenorrhea pain severity. We used a **linear regression model** adjusting for age and body mass index, identifying one genome-wide significant (P < 5 × 10−8) association (rs7523086, P = 4.1 × 10−14, effect size 0.1 [95% confidence interval, 0.074–0.126]). This single nucleotide polymorphism is colocalising with NGF, encoding nerve growth factor.

**Results 3.2. Results from genome-wide association study on dysmenorrhea pain severity**

&emsp;&emsp;To identify common genetic factors that correlate with dysmenorrhea pain severity, we analysed participant data from a total of 11,892 unrelated females of European descent. After quality controls, a total of 11,942,402 single nucleotide polymorphisms (SNPs) were analysed by linear regression adjusting for BMI, age, and principal components, and then adjusted for a genomic inflation factor of 1.024. Analysis revealed one genome-wide significant (GWS) association (P < 5 × 10−8) at a region on chromosome 1p13.2. The GWS association colocalises with NGF, the gene encoding nerve growth factor (rs7523086, P = 4.1 × 10−14). The index SNP rs7523086 is a common polymorphism (G/A, MAF A allele: 0.328, 1000 Genomes CEU population), and the effect size of 0.1 (95% confidence interval, 0.074-0.126), means that the presence of one risk allele corresponds to a predicted increase of 0.1 points in a 4-point ordinal scale for increasing pain intensity. Furthermore, there were 5 loci with suggestive levels of association (P < 1 × 10−7).

**Results 3.3. Functional annotation of genetic association at 1p13.2**

&emsp;&emsp;We investigated the functional significance of the GWS association at 1q13.2 by using the Web-based computational tool HaploReg, which facilitates identification of tagging SNPs in high linkage disequilibrium, and exploration of chromatin states, conservations, and transcription factor regulatory motif alterations.

&emsp;&emsp;We identified 23 variants in high linkage disequilibrium (r2 > 0.8) with the GWS index SNP rs7523086, which cover an approximate 23-kb region 5′ to NGF, with one variant situated in an NGF intron. Moreover, our nominated SNPs and colocalising marks of histone activation span a region that encompass the last intron of NGF, and the predicted transcriptional start site of a noncoding RNA, called RP4-663N10.1. This noncoding RNA produces a 2,610-bp transcript from 2 exons read in the forward orientation. The entire NGF locus sits within the single intron of RP4-663N10.1 and is transcribed in the opposite, antisense direction.

&emsp;&emsp;**Inspection of RP4-663N10.1 expression across the range of tissues captured by GTEx15,20,21 demonstrated highest expression in adipose, aorta, and various gynaecological tissues, including the uterus, fallopian tube, and cervix, with no/little expression detected in brain tissues or peripheral blood leukocytes. This expression profile closely mirrors that observed for NGF, although RP4-663N10.1 expression levels are vastly smaller in magnitude compared with that for NGF (Fig. 5). Interestingly, many of the pain severity variants in high linkage with the index SNP were identified as expression quantitative loci (eQTL) for RP4-663N10.1 (Fig. 5). Results from the GTEx Project, which examines genotypic correlation with tissue-specific gene expression levels in samples from healthy male and female donors,15,20,21 indicated the minor allele of the variant rs6328 (r2 = 0.85 with dysmenorrhea index SNP rs7523086) is significantly correlated with increased RP4-663N10.1 expression in aorta tissue (P = 1.8 × 10−7, effect size 0.34).**
